server/src/main/java/org/spine3/server/projection/ProjectionRepository.java

```diff
@@ -287,20 +359,53 @@ public void setOnline() {
         setStatus(Status.ONLINE);
     }
 
-   /**
+    private void completeCatchUp() {
+        setOnline();
+    }
+
+    private boolean isBulkWriteInProgress() {
+        return ongoingOperation != null
+                && ongoingOperation.isInProgress();
+    }
+
+    private boolean isBulkWriteRequired() {
+        return !catchUpMaxDuration.equals(
+                catchUpMaxDuration.getDefaultInstanceForType());
+    }
+
+    /**
      * The stream observer which redirects events from {@code EventStore} to
      * the associated {@code ProjectionRepository}.
      */
     private static class EventStreamObserver implements StreamObserver<Event> {
 
         private final ProjectionRepository projectionRepository;
+        private BulkWriteOperation operation;
 
         EventStreamObserver(ProjectionRepository projectionRepository) {
             this.projectionRepository = projectionRepository;
         }
 
         @Override
         public void onNext(Event event) {
+            if (projectionRepository.status != Status.CATCHING_UP) {
```

Comment:
@alexander-yevsyukov Now I see the same thing as you do, but this must an issue of the `codecov` browser extension. The `IDEA` and [codecove](https://codecov.io/gh/SpineEventEngine/core-java/src/projection-repo-catchup-performance/server/src/main/java/org/spine3/server/projection/ProjectionRepository.java) both display that this method is covered appropriately.